trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/standardfunctions/StandardFunctionHandler.java

```diff
@@ -139,86 +69,70 @@
-	 * get a new nondeterministic value even if the variable occurs in a loop. If this constant is set, we havoc the
-	 * variable also before the nondeterministic assignment. If the auxvar is also havoced before, it is only
-	 * backward-live to the havoc, otherwise it would be backward-live until the beginning of the procedure.
-	 */
-	private static final boolean HAVOC_NONDET_AUXVARS_ALSO_BEFORE = true;
-
-	/**
-	 * See MEMORY_ORDER_SEQ_CST in stdatomic.h
-	 */
-	private static final int MEMORY_ORDER_SEQ_CST = 5;
-
 	private final LocationFactory mLocationFactory;
-
 	private final Map<String, IFunctionModelHandler> mFunctionModels;
-
-	private final ExpressionTranslation mExpressionTranslation;
-
-	private final MemoryHandler mMemoryHandler;
-
-	private final TypeSizeAndOffsetComputer mTypeSizeComputer;
-
-	private final ProcedureManager mProcedureManager;
-
-	private final CTranslationResultReporter mReporter;
-
 	private final Map<String, IASTNode> mFunctionTable;
-
-	private final AuxVarInfoBuilder mAuxVarInfoBuilder;
-
-	private final INameHandler mNameHandler;
-
-	private final TypeSizes mTypeSizes;
-
 	private final FlatSymbolTable mSymboltable;
-
 	private final TranslationSettings mSettings;
-
-	private final ExpressionResultTransformer mExprResultTransformer;
-
-	private final ITypeHandler mTypeHandler;
-
-	private final CExpressionTranslator mCEpressionTranslator;
-
-	private final ThreadIdManager mThreadIdManager;
-
-	private final DataRaceChecker mDataRaceChecker;
-
 	private final ILogger mLogger;
 
 	public StandardFunctionHandler(final ILogger logger, final Map<String, IASTNode> functionTable,
 			final AuxVarInfoBuilder auxVarInfoBuilder, final INameHandler nameHandler,
 			final ExpressionTranslation expressionTranslation, final MemoryHandler memoryHandler,
 			final TypeSizeAndOffsetComputer typeSizeAndOffsetComputer, final ProcedureManager procedureManager,
-			final CTranslationResultReporter reporter, final TypeSizes typeSizes, final FlatSymbolTable symboltable,
-			final TranslationSettings settings, final ExpressionResultTransformer expressionResultTransformer,
-			final LocationFactory locationFactory, final ITypeHandler typeHandler,
-			final CExpressionTranslator cEpressionTranslator, final DataRaceChecker dataRaceChecker) {
+			final TypeSizes typeSizes, final FlatSymbolTable symboltable, final TranslationSettings settings,
+			final ExpressionResultTransformer expressionResultTransformer, final LocationFactory locationFactory,
+			final ITypeHandler typeHandler, final CExpressionTranslator cEpressionTranslator,
+			final DataRaceChecker dataRaceChecker) {
+		final List<FunctionModelProvider> functionModelProviders = List.of(
+				new LinuxFunctionModelProvider(auxVarInfoBuilder, nameHandler, expressionTranslation, memoryHandler,
+						typeSizeAndOffsetComputer, procedureManager, typeSizes, settings, expressionResultTransformer,
+						typeHandler, cEpressionTranslator, dataRaceChecker),

```

Comment:
It looks like we could use some container object for the different FunctionModelProvider

Reply:
The list of all these objects was now moved into a method of [CHandler](https://github.com/ultimate-pa/ultimate/pull/714/files#diff-eac2932341760ee7522f505fdef04efb90ec6eaf8e279341fb02af1b032730e1R609-R637). Do you still think, we should move this to a separate object?